tests/integrationv2/test_pq_handshake.py

```diff
@@ -32,124 +26,35 @@
-            "kem": None, "kem_group": None},
-    (Ciphers.KMS_PQ_TLS_1_0_2020_07, Ciphers.KMS_PQ_TLS_1_0_2020_07):
-        {"cipher": "ECDHE-KYBER-RSA-AES256-GCM-SHA384",
-            "kem": "kyber512r3", "kem_group": None},
-
-    (Ciphers.PQ_SIKE_TEST_TLS_1_0_2019_11, Ciphers.KMS_PQ_TLS_1_0_2019_06):
-        {"cipher": "ECDHE-RSA-AES256-GCM-SHA384",
-            "kem": None, "kem_group": None},
-    (Ciphers.PQ_SIKE_TEST_TLS_1_0_2019_11, Ciphers.KMS_PQ_TLS_1_0_2020_02):
-        {"cipher": "ECDHE-RSA-AES256-GCM-SHA384",
-            "kem": None, "kem_group": None},
-    (Ciphers.PQ_SIKE_TEST_TLS_1_0_2019_11, Ciphers.KMS_PQ_TLS_1_0_2020_07):
-        {"cipher": "ECDHE-RSA-AES256-GCM-SHA384",
-            "kem": None, "kem_group": None},
-
-    (Ciphers.PQ_SIKE_TEST_TLS_1_0_2020_02, Ciphers.KMS_PQ_TLS_1_0_2019_06):
-        {"cipher": "ECDHE-RSA-AES256-GCM-SHA384",
-            "kem": None, "kem_group": None},
-    (Ciphers.PQ_SIKE_TEST_TLS_1_0_2020_02, Ciphers.KMS_PQ_TLS_1_0_2020_02):
-        {"cipher": "ECDHE-RSA-AES256-GCM-SHA384",
-            "kem": None, "kem_group": None},
-    (Ciphers.PQ_SIKE_TEST_TLS_1_0_2020_02, Ciphers.KMS_PQ_TLS_1_0_2020_07):
-        {"cipher": "ECDHE-RSA-AES256-GCM-SHA384",
-            "kem": None, "kem_group": None},
-
-    (Ciphers.KMS_PQ_TLS_1_0_2019_06, Ciphers.KMS_TLS_1_0_2018_10):
-        {"cipher": "ECDHE-RSA-AES256-GCM-SHA384",
-            "kem": None, "kem_group": None},
-    (Ciphers.KMS_PQ_TLS_1_0_2020_02, Ciphers.KMS_TLS_1_0_2018_10):
-        {"cipher": "ECDHE-RSA-AES256-GCM-SHA384",
-            "kem": None, "kem_group": None},
-    (Ciphers.KMS_PQ_TLS_1_0_2020_07, Ciphers.KMS_TLS_1_0_2018_10):
-        {"cipher": "ECDHE-RSA-AES256-GCM-SHA384",
-            "kem": None, "kem_group": None},
-
-    (Ciphers.KMS_TLS_1_0_2018_10, Ciphers.KMS_PQ_TLS_1_0_2019_06):
-        {"cipher": "ECDHE-RSA-AES256-GCM-SHA384",
-            "kem": None, "kem_group": None},
-    (Ciphers.KMS_TLS_1_0_2018_10, Ciphers.KMS_PQ_TLS_1_0_2020_02):
-        {"cipher": "ECDHE-RSA-AES256-GCM-SHA384",
-            "kem": None, "kem_group": None},
-    (Ciphers.KMS_TLS_1_0_2018_10, Ciphers.KMS_PQ_TLS_1_0_2020_07):
-        {"cipher": "ECDHE-RSA-AES256-GCM-SHA384",
-            "kem": None, "kem_group": None},
-
-    # The expected kem_group string for this case purposefully excludes a curve;
-    # depending on how s2n was compiled, the curve may be either x25519 or one
-    # of the NIST curves.
-    (Ciphers.PQ_TLS_1_0_2020_12, Ciphers.PQ_TLS_1_0_2020_12):
-        {"cipher": "TLS_AES_256_GCM_SHA384",
-            "kem": None, "kem_group": "_kyber-512-r3"},
-    (Ciphers.PQ_TLS_1_0_2020_12, Ciphers.PQ_TLS_1_0_2023_01):
-        {"cipher": "TLS_AES_256_GCM_SHA384",
-            "kem": None, "kem_group": "_kyber-512-r3"},
+    # The tuple keys have the form:
+    # (client_{cipher, kem_group}, server_{cipher, kem_group}): {"cipher": {expected_cipher}, "kem_group": {expected_kem_group}}
```

Comment:
What does the key formatting here mean? The "{cipher, kem_group}" part? Isn't it just:
```suggestion
    # (client_cipher, server_cipher): {"cipher": {expected_cipher}, "kem_group": {expected_kem_group}}
```

Reply:
Some of them are "Ciphers" (aka s2n Security Policies), others are "KemGroups" (aka AWS-LC/Boring/OpenSSL Supported Groups). 

`(Ciphers.PQ_TLS_1_0_2023_01, Ciphers.PQ_TLS_1_0_2023_01)` is a `(Client_Cipher, Server_Cipher)`. 

`(KemGroups.P384_KYBER768R3, Ciphers.PQ_TLS_1_3_2023_06_01)` is `(Client_KemGroup, Server_Cipher)`.

`(Ciphers.PQ_TLS_1_3_2023_06_01, KemGroups.X25519Kyber768Draft00)` is `(Client_Cipher, Server_KemGroup)`.

Reply:
Oh, THAT'S what it's trying to communicate! I wonder if it'd be clearer if it was "client_{cipher|kem_group}", or even just "client". Other places in the same notation we're currently using "," as a list, not an "or".

Reply:
Part of the confusion might also come from the fact that both security policies and kem groups are labeled "ciphers". So we're being specific that some ciphers are actually kem groups, but not that the other ciphers are actually security policies, and the same definition also uses "cipher" as an actual cipher suite.